src/zimscraperlib/imaging.py

```diff
@@ -40,21 +40,40 @@ def solarize(r, g, b):
     return rgb_to_hex(mr, mg, mb), rgb_to_hex(sr, sg, sb)
 
 
-def resize_image(fpath, width, height=None, to=None, method="width"):
+def resize_image(
+    fpath, width, height=None, to=None, method="width", allow_upscaling=True
+):
     """ resize an image file (dimensions)
 
-        methods: width, height, cover """
+        methods: width, height, cover
+        allow upscaling: upscale image preserving aspect ratio if required before resizing """
     with open(str(fpath), "rb") as fp:
-        with PIL.Image.open(fp) as image:
-            if method == "width":
-                resized = resizeimage.resize(method, image, width)
-            elif method == "height":
-                resized = resizeimage.resize(method, image, height)
-            else:
-                resized = resizeimage.resize(method, image, [width, height])
+        image = PIL.Image.open(fp)
+        # prserve image format
+        image_format = image.format
+
+        # upscale if required preserving the aspect ratio
+        if allow_upscaling:
+            height_width_ratio = float(image.size[1]) / float(image.size[0])
+            if image.size[0] < width:
+                image = image.resize((width, int(width * height_width_ratio)))
+            if height and image.size[1] < height:
+                image = image.resize((int(height / height_width_ratio), height))
+
+        # resize using the requested method
+        if method == "width":
+            resized = resizeimage.resize(method, image, width)
+        elif method == "height":
+            resized = resizeimage.resize(method, image, height)
+        else:
+            resized = resizeimage.resize(method, image, [width, height])
+
+    # save the image
+    if resized.mode == "RGBA" and image_format == "JPEG":
```

Comment:
Again, introducing new code without explaining it. What's this for?
What you are doing here is allowing to use this function as a way to convert from one format that use RGBA (png) into JPEG without crashing.
This is not the right way to do it as you will not crash and convert but you won't make use of the `kwargs` stuff bellow.

I think it would be handy to be able to convert from one format to another via the lib but I don't like the idea of using this function to.

We can add another convert method to imaging in this PR if you feel like it ; should be relatively easy. Users would want to both convert and resize can make two calls or we have an additional shortcut but we should keep functions dedicated and small.

Reply:
Yup. I'm doing another method for this. BTW, I never wanted this to be a do-it-all function. I added this as while using contain mode, the image mode changes from RGB to RGBA. I'll add kwargs to it. Previously it worked in tests because we used png_image always and the format was PNG even when saving JPEG

Reply:
Yes, that's actually quite bad as you'd get an RGB png, loosing transparency in the process.
Pillow is so flexible it's tempting to do it all at once but I think it will lead to difficulties in the past.
Conversion would be a good addition. thanks